client/src/pages/Explore.js

```diff
@@ -1,98 +1,181 @@
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "flex-end",
+    overflow: "hidden",
+  },
+  title: {
+    margin: 5,
+    textAlign: "center",
+  },
+  customCheckBoxRoot: {
+    width: 5,
+    height: 5,
+  },
+  bottomInformationContainer: {
+    display: "flex",
+    height: "15%",
+    borderTop: "1px solid #a9a9a9",
+    padding: 10,
+    justifyContent: "space-between",
+  },
+  bottomInformationSubContainer1: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    paddingLeft: 10,
+  },
+  bottomInformationSubContainer2: {
+    display: "flex",
+    flexDirection: "column",
+    justifyContent: "center",
+    alignItems: "flex-end",
+  },
+  subtitle2: {
+    color: "#c5bec4",
+    fontSize: 12,
+  },
+  favoriteDefaultIcon:{
+    color: "white"
+  },
+  favoriteCheckedIcon:{
+    color: "orange"
+  },
+  gridContainer: {
+    marginTop: 32, height: "75%"
+  },
+  legend1: {
+    fontSize: 17,
+    color: "white"
+  },
+  legend2:{
+    fontSize: 11,
+    color: "rgb(175 175 175)"
+  }
 });
 
-function FavoriteCheckBox() {
+function FavoriteCheckBox({ place, handleFavoriteChange }) {
   const classes = useStyles();
-  const [checked, setChecked] = useState(false);
+  const [checked, setChecked] = useState(place.favorite);
   return (
-    <div>
+    <>
       <CustomSmallerCheckBox
         checked={checked}
-        onChange={(e) => setChecked(e.target.checked)}
-        icon={<Favorite style={{ color: "white" }} />}
-        checkedIcon={<Favorite style={{ color: "orange" }} />}
+        onChange={(e) => {
+          handleFavoriteChange(e.target.checked, place.name);
+          setChecked(e.target.checked);
+        }}
+        icon={<Favorite className={classes.favoriteDefaultIcon} />}
+        checkedIcon={<Favorite className={classes.favoriteCheckedIcon} />}
         classes={{ root: classes.customCheckBoxRoot }}
       />
-    </div>
+    </>
   );
 }
 
+//@TODO: This userId should come from props 
+const userId = "abcd123";
 const Explore = () => {
   const classes = useStyles();
+  const [favorites, setFavorites] = useState([]);
+  const [places, setPlaces] = useState([]);
+  useEffect(() => {
+    async function getData() {
+      try {
+        const favoriteResponse = await (await fetch(
+          `/api/users/${userId}/favorite-cities`,
+          {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        )).json();
+        //Assuming that the explore page can be accessed be both authenticated and unAuthenticated user.
+        const favoriteList = Array.isArray(favoriteResponse) ? favoriteResponse : [];
+        setFavorites(favoriteList);
+        const cityList = await (
+          await fetch("/api/cities", {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          })
+        ).json()
```

Comment:
this would be clearer to read as something like:
```javascript
const cityListResponse = await fetch("/api/cities", {
  method: "GET",
  headers: {
    "Content-Type": "application/json",
  },
});
const cityList = await cityListResponse.json();
```